lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp

```diff
@@ -253,7 +253,7 @@ namespace Js
 
         if (info.GetPropertyString() != nullptr && info.GetPropertyString()->ShouldUseCache())
         {
-            Assert(propertyString == info.GetPropertyString());
+            Assert(info.IsNoCache() || propertyString == info.GetPropertyString());
```

Comment:
Ok, just that the new condition is duplicated :)

---
In reply to: [126219071](https://github.com/Microsoft/ChakraCore/pull/3153#discussion_r126219071) [](ancestors = 126219071)